lua/lvim/core/theme.lua

```diff
@@ -0,0 +1,49 @@
+local M = {}
+
+M.config = function()
+  lvim.builtin.theme = {
+    name = "tokyonight",
+    options = {
+      style = "night", -- The theme comes in three styles, `storm`, a darker variant `night` and `day`
+      transparent = false, -- Enable this to disable setting the background color
+      terminal_colors = true, -- Configure the colors used when opening a `:terminal` in Neovim
+      styles = {
+        -- Style to be applied to different syntax groups
+        -- Value is any valid attr-list value for `:help nvim_set_hl`
+        comments = { italic = true },
+        keywords = { italic = true },
+        functions = {},
+        variables = {},
+        -- Background styles. Can be "dark", "transparent" or "normal"
+        sidebars = "dark", -- style for sidebars, see below
+        floats = "dark", -- style for floating windows
+      },
+      sidebars = { "qf", "help" }, -- Set a darker background on sidebar-like windows. For example: `["qf", "vista_kind", "terminal", "packer"]`
+      day_brightness = 0.3, -- Adjusts the brightness of the colors of the **Day** style. Number between 0 and 1, from dull to vibrant colors
+      hide_inactive_statusline = false, -- Enabling this option, will hide inactive statuslines and replace them with a thin border instead. Should work with the standard **StatusLine** and **LuaLine**.
+      dim_inactive = false, -- dims inactive windows
+      lualine_bold = false, -- When `true`, section headers in the lualine theme will be bold
+
+      --- You can override specific color groups to use other groups or a hex color
+      --- fucntion will be called with a ColorScheme table
+      on_colors = function(_colors) end,
+
+      --- You can override specific highlights to use other groups or a hex color
+      --- fucntion will be called with a Highlights and ColorScheme table
+      on_highlights = function(_highlights, _colors) end,
+      use_background = true, -- can be light/dark/auto. When auto, background will be set to vim.o.background
+    },
+  }
+end
+
+M.setup = function()
+  local status_ok, theme = pcall(require, "tokyonight")
+  if not status_ok then
+    return
+  end
+
+  theme.setup(lvim.builtin.theme.options)
+  lvim.builtin.lualine.options.theme = "tokyonight"
```

Comment:
```suggestion
  local selected_theme =  lvim.builtin.theme.name
  local status_ok, plugin = pcall(require, selected_theme)
  if not status_ok then
    return
  end

  plugin.setup(lvim.builtin.theme.options)
  lvim.builtin.lualine.options.theme = selected_theme
```

Reply:
lualine only supports `tokyonight`
```
To enable the TokyoNight theme for Lualine, simply specify it in your lualine settings:

require('lualine').setup {
  options = {
    -- ... your lualine config
    theme = 'tokyonight'
    -- ... your lualine config
  }
}
```
it doesn't support the flavors

Reply:
also if we set the `lvim.builtin.theme.style` we don't need to load other themes ( at least I think we don't 🤔 )

```
  style = "storm", -- The theme comes in three styles, `storm`, `moon`, a darker variant `night` and `day`
```

Reply:
> do you mean using `lvim.colorscheme.options` instead of `lvim.builtin.theme.options` ?

I didn't see that one in time, I would just deprecating `lvim.colorscheme` if you like this suggsetion above^

Reply:
> lualine only supports `tokyonight`

are we not talking about these themes?

![image](https://user-images.githubusercontent.com/59826753/191199174-7638cb29-abae-4e19-8fca-74462017d4f9.png)

Reply:
hmm, it is too much of a breaking change imo,
 lets keep the change to a minimum, like probably nobody would even bother changing `lvim.builtin.theme.options` 😅

Reply:
@kylo252 no we are talking about this theme -> https://github.com/folke/tokyonight.nvim/tree/main/lua/lualine/themes

Reply:
> > lualine only supports `tokyonight`
> 
> are we not talking about these themes?
> 

@kylo252 no we are talking about this theme -> https://github.com/folke/tokyonight.nvim/tree/main/lua/lualine/themes

Reply:
> hmm, it is too much of a breaking change imo,

sounds good!